fxgl-core/src/main/java/com/almasb/fxgl/core/collection/grid/Grid.java

```diff
@@ -136,6 +157,15 @@ public final List<T> getNeighbors(int x, int y) {
         getUp(x, y).ifPresent(result::add);
         getRight(x, y).ifPresent(result::add);
         getDown(x, y).ifPresent(result::add);
+
+        // Include "Corner" neighbors only if diagonal movement is allowed
```

Comment:
This should go into a new method `getNeighbors` that allows us to provide info on how to extract the neighbors.

```
getNeighbors(x, y, NeighborFilteringOption.FOUR_DIRECTIONS)

NeighborFilteringOption.EIGHT_DIRECTIONS
```
If we need more in the future, we can modify as appropriate.